waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -29,7 +29,7 @@ logScope:
 type
   PeerManager* = ref object of RootObj
     switch*: Switch
```

Comment:
Looking forward to having the `switch` removed here in future (not this PR, of course :))

Reply:
Right, from the perspective of connection management it does make sense to have a `switch` as dialer. That said, I'm still leaning towards separating these two paradigms: a peer manager that operates on a peer store (and provides, e.g. API for discovery methods to add peers) and a connection manager that reads from said peer store and uses the `switch` as dialer to maintain certain connectivity goals. If not physically separated into separate modules it may still makes sense to have these logically separate?

Would `PeerPool` be roughly equivalent to what I described as the peer manager above?

Reply:
> a peer manager that operates on a peer store (and provides, e.g. API for discovery methods to add peers)

What I had in mind is that the methods for adding peers are part of the PeerStore and not the PeerManager. The source is specified when adding the peer, and can be bootstrap/discv5 or other.

Will raise this topic in the meeting we have pending, to gather everyone's input.

> Would PeerPool be roughly equivalent to what I described as the peer manager above?

* nimbus: PeerPool is similar to our peerstore, but doesn't maintain any connectivity goals. It allows to for example [addPeer](https://github.com/status-im/nimbus-eth2/blob/stable/beacon_chain/networking/peer_pool.nim#L462) but has no knowledge of discovery. Connectivity goals are handled in eth2_network.nim.
* nim-eth: PeerPool both stores peers and connects to them, for example, [trying to connect to more peers](https://github.com/status-im/nim-eth/blob/master/eth/p2p/peer_pool.nim#L148) if below minPeers. [node disvoery](https://github.com/status-im/nim-eth/blob/master/eth/p2p/peer_pool.nim#L103) is inside the peer pool.

Reply:
Ok, makes sense to me! We can discuss functionality in more detail.